connect/mirror/src/test/java/org/apache/kafka/connect/mirror/integration/MirrorConnectorsWithCustomForwardingAdminIntegrationTest.java

```diff
@@ -0,0 +1,596 @@
+        waitForTopicCreated(backup, "primary.heartbeats");
+
+        // expect to use FakeForwardingAdminWithLocalMetadata to create remote topics into local store
+        assertTrue(FakeLocalMetadataStore.allTopics.containsKey("primary.test-topic-1"));
+        assertTrue(FakeLocalMetadataStore.allTopics.containsKey("backup.test-topic-1"));
+
+        // expect to use FakeForwardingAdminWithLocalMetadata to create internal topics into local store
+        assertTrue(FakeLocalMetadataStore.allTopics.containsKey("heartbeats"));
+        assertTrue(FakeLocalMetadataStore.allTopics.containsKey("backup.heartbeats"));
+        assertTrue(FakeLocalMetadataStore.allTopics.containsKey("primary.heartbeats"));
+        assertTrue(FakeLocalMetadataStore.allTopics.containsKey("backup.checkpoints.internal"));
+        assertTrue(FakeLocalMetadataStore.allTopics.containsKey("primary.checkpoints.internal"));
+        assertTrue(FakeLocalMetadataStore.allTopics.containsKey("mm2-offset-syncs.primary.internal"));
+        assertTrue(FakeLocalMetadataStore.allTopics.containsKey("mm2-offset-syncs.backup.internal"));
+    }
+
+    @Test
+    public void testCreatePartitionsUseProvidedForwardingAdmin() throws Exception {
+        mm2Config = new MirrorMakerConfig(mm2Props);
+
+        waitUntilMirrorMakerIsRunning(backup, CONNECTOR_LIST, mm2Config, PRIMARY_CLUSTER_ALIAS, BACKUP_CLUSTER_ALIAS);
+        waitUntilMirrorMakerIsRunning(primary, CONNECTOR_LIST, mm2Config, BACKUP_CLUSTER_ALIAS, PRIMARY_CLUSTER_ALIAS);
+
+        // make sure the topic is auto-created in the other cluster
+        waitForTopicCreated(primary, "backup.test-topic-1");
+        waitForTopicCreated(backup, "primary.test-topic-1");
+
+        // expect to use FakeForwardingAdminWithLocalMetadata to create remote topics into local store
+        assertTrue(FakeLocalMetadataStore.allTopics.containsKey("primary.test-topic-1"));
+        assertTrue(FakeLocalMetadataStore.allTopics.containsKey("backup.test-topic-1"));
+        assertEquals(FakeLocalMetadataStore.allTopics.get("primary.test-topic-1").get("partitions"), String.valueOf(NUM_PARTITIONS));
+        assertEquals(FakeLocalMetadataStore.allTopics.get("backup.test-topic-1").get("partitions"), String.valueOf(NUM_PARTITIONS));
+
+        // increase number of partitions
+        HashMap<String, NewPartitions> newPartitions = new HashMap<String, NewPartitions>() {{
+            put("test-topic-1", NewPartitions.increaseTo(NUM_PARTITIONS + 1));
+        }};
+
+        primary.kafka().createAdminClient(Utils.mkProperties(superUserConfig())).createPartitions(newPartitions).all().get();
+        waitForTopicPartitionCreated(backup, "primary.test-topic-1", NUM_PARTITIONS + 1);
+
+        // expect to use FakeForwardingAdminWithLocalMetadata to update number of partitions in local store
+        assertEquals(FakeLocalMetadataStore.allTopics.get("primary.test-topic-1").get("partitions"), String.valueOf(NUM_PARTITIONS + 1));
```

Comment:
Should we swap the order of the arguments? `String.valueOf(NUM_PARTITIONS + 1)` seems like the expected value here. Same below

Reply:
done